docs/release_flow.md

```diff
@@ -0,0 +1,62 @@
+# About Release FLow
```

Comment:
Please add a path/reference in the root README.md, so that it is easier for users to find this doc